src/scripts/main.js

```diff
@@ -1 +1,80 @@
 'use strict';
+
+const firstPromise = () => {
+  return new Promise((resolve, reject) => {
+    document.body.addEventListener('click', () => {
+      resolve('First promise was resolved');
+    });
+
+    setTimeout(() => {
+      reject(new Error('First promise was rejected'));
+    }, 3000);
```

Comment:
The first promise has a potential issue where both the click event and the timeout could trigger resolve and reject simultaneously. Consider removing the event listener after it resolves or rejects to prevent multiple executions.